### PR TITLE
Archiving of scenes from S3, added backlog processing DAG

### PR DESCRIPTION
**Pull request by sachitrajbhandari**
https://gajira.atlassian.net/secure/RapidBoard.jspa?rapidView=549&selectedIssue=DEADM-1407

### CODE REVIEW

File: scripts/c3_to_s3_rolling.py [hunk 1]
```diff
@@ -755,52 +669,110 @@ def sync_granules(file_path, nci_dir, s3_root_path, s3_bucket, s3_base_url, sns_
 
     # For each granule, sync it if it needs syncing
     if granules_count > 0:
-        for metadata_file in list_of_granules:
-            LOG.info(f"Processing {metadata_file}")
+        for granule_row in list_of_granules:
 
-            metadata_file_path = Path(metadata_file)
+            metadata_file = granule_row[0] if len(granule_row) > 0 else None
+            archived_date = granule_row[1] if len(granule_row) > 1 else None
```
Comment (alexgleith) [inline scripts/c3_to_s3_rolling.py:675]: This doesn't store a date, but a flag. It probably should be `archived` as True or False.

File: scripts/c3_to_s3_rolling.py [hunk 2]
```diff
@@ -755,52 +669,110 @@ def sync_granules(file_path, nci_dir, s3_root_path, s3_bucket, s3_base_url, sns_
...
+                            f"Failed to archive {granule} "
+                            f"because of an error in the rm command - {exp}"
+                        )
                 else:
-                    LOG.warning(f"Metadata exists in S3 and update is not set to True, "
-                                f"not syncing {granule}")
-
+                    LOG.warning(
+                        f"Metadata doesn't exists in S3, "
+                        f"not archiving {granule}"
```
Comment (alexgleith) [inline scripts/c3_to_s3_rolling.py:729]:
```suggestion
                        f"not deleting anything from S3 for {granule}"
```

File: dags/nci_c3_upload_s3_backlog.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    
+    # Load the latest stable DEA module
+    module use /g/data/v10/public/modules/modulefiles
+    module load dea
+    
+    # Be verbose and echo what we run
+    set -x
+    
+    args="-h 130.56.244.105 datacube -t -A -F,"
+    query="SELECT dsl.uri_body, ds.archived
```
[the remark this answers is not in the document]
Comment (sachitrajbhandari) [reply dags/nci_c3_upload_s3_backlog.py:56]: Yes you are right. Revised SQL.

File: dags/nci_c3_upload_s3.py [hunk 4]
```diff
@@ -13,13 +13,15 @@
 
  * `s3bucket`: Name of the S3 bucket. `"dea-public-data"`
  * `s3baseurl`: Base URL of the S3. `"https://data.dea.ga.gov.au/"`
+ * `explorertbaseurl`: Base URL of the explorer. `"https://explorer.prod.dea.ga.gov.au"`
```
[the remark this answers is not in the document]
Comment (sachitrajbhandari) [reply dags/nci_c3_upload_s3.py]: Fixed

File: dags/nci_c3_upload_s3.py [hunk 5]
```diff
@@ -13,13 +13,15 @@
...
  * `snstopic`: ARN of the SNS topic. `"arn:aws:sns:ap-southeast-2:451924316694:landsat-collection-3-dev-topic"`
  * `doupdate`: If this flag is set then do a fresh sync of data and
     replace the metadata. `"--force-update"`
 
     {
         "s3bucket": "dea-public-data-dev",
         "s3baseurl": "http://dea-public-data-dev.s3-ap-southeast-2.amazonaws.com",
+        "explorertbaseurl": "https://explorer.prod.dea.ga.gov.au",
```
[the remark this answers is not in the document]
Comment (sachitrajbhandari) [reply dags/nci_c3_upload_s3.py]: Fixed

File: scripts/c3_to_s3_rolling.py (hunk 1 above)
Comment (sachitrajbhandari) [reply scripts/c3_to_s3_rolling.py]: Replace with flag.

File: scripts/c3_to_s3_rolling.py (hunk 2 above)
Comment (sachitrajbhandari) [reply scripts/c3_to_s3_rolling.py]: Revised

**Pull request merged**